### PR TITLE
fix: gphdfs unicode support

### PR DESCRIPTION
**Pull request by sshang**
SIGSEGV on segment when reading data form avro file that contains unicode
character.  In this case the character was this
http://www.fileformat.info/info/unicode/char/85/index.htm

### CODE REVIEW

File: gpAux/extensions/gphdfs/src/java/1.2/com/emc/greenplum/gpdb/hadoop/io/GPDBWritable.java [hunk 1]
```diff
@@ -418,8 +418,8 @@ else if (!isTextForm(colType[i]) && coldbtype.isVarLength())
 					 * at the beginning and add a "\0" at the end */
 					default: {
 						String outStr = (String)colValue[i]+"\0";
-						out.writeInt(outStr.length());
 						byte[] data = (outStr).getBytes(CHARSET);
+						out.writeInt(data.length);
 						out.write(data);
```
Comment (yydzero) [inline gpAux/extensions/gphdfs/src/java/1.2/com/emc/greenplum/gpdb/hadoop/io/GPDBWritable.java]: Please add some cases.

Comment (sshang) [inline gpAux/extensions/gphdfs/src/java/1.2/com/emc/greenplum/gpdb/hadoop/io/GPDBWritable.java]: sure. will add cases in another story

**Comment by sshang** [comment]
@hlinnaka , String in JAVA is encoded by unicode, that means, '\u0085' is one char, but it has two bytes.
i.e. you can run this code in java, the output is 1.
		String uni = "\u0085";
		System.out.println(uni.length());

**Comment by hlinnaka** [comment]
Right, but ẃhy does that lead to a segfault?

**Comment by sshang** [comment]
the gphdfs protocol is : lengh + byte[length], is length is wrong, gpdb reads wrong numbers of data, and may cause a segfault.

**Comment by hlinnaka** [comment]
On 11/16/2015 01:06 PM, sshang wrote:
> the gphdfs protocol is : lengh + byte[length], is length is wrong, gpdb reads wrong numbers of data, and may cause a segfault.

Ok. It shouldn't. Please also add suitable safeguards or sanity checks 
in the server-side code to not segfault on bad input like that.

- Heikki

**Comment by hlinnaka** [comment]
@sshang, is this PR still relevant? Please close if not.

**Comment by hlinnaka** [comment]
Closing as there hasn't been any activity on this for a long time.

**Pull request closed without merging**